To: vim_dev@googlegroups.com
Subject: Patch 7.3.372
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.372
Problem:    When using a command line mapping to <Up> with file name 
            completion to go one directory up, 'wildchar' is inserted.
            (Yasuhiro Matsumoto) 
Solution:   Set the KeyTyped flag.
Files:      src/ex_getln.c


*** ../vim-7.3.371/src/ex_getln.c	2011-10-26 22:02:10.000000000 +0200
--- src/ex_getln.c	2011-12-08 18:40:49.000000000 +0100
***************
*** 645,651 ****
--- 645,655 ----
  		}
  		else if (ccline.cmdpos > i)
  		    cmdline_del(i);
+ 
+ 		/* Now complete in the new directory. Set KeyTyped in case the
+ 		 * Up key came from a mapping. */
  		c = p_wc;
+ 		KeyTyped = TRUE;
  	    }
  	}
  
*** ../vim-7.3.371/src/version.c	2011-12-08 17:49:31.000000000 +0100
--- src/version.c	2011-12-08 18:42:54.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     372,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
255. You work for a newspaper and your editor asks you to write an
     article about Internet addiction...in the "first person."

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
